### PR TITLE
add SEP-0011 black

### PR DESCRIPTION
**Pull request by nabobalis**

### CODE REVIEW

File: SEP-0011.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
+# SEP-0011 -- Formatting Code with *Black*
```
Comment (Cadair) [inline SEP-0011.md:1]: There definitely needs to be an attribution section in this somewhere... :wink:

File: SEP-0011.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+Note that ideas and most text for this SEP have been adapted from a similar enhancement proposal from [Django DEP-0008](https://github.com/django/deps/blob/main/final/0008-black.rst) and [Astropy APE 20](https://github.com/astropy/astropy-APEs/blob/main/APE20.rst).
+
+#### Rationale
+
+**Consistent and readable code formatting** reduces effort in both code review and in future maintenance.
+Code is read far more often than it is written, so it makes sense to invest effort to ensure that the code is easy to read later.
+
+Ensuring that code formatting meets ``sunpy`` standards is currently done by applying CI checks based on PEP-8.
+Learning and consistently applying all these rules is not easy.
+There are at least 200 commits in ``sunpy`` which were solely address PEP-8 violations, so this is a real issue.
```
Comment (Cadair) [inline SEP-0011.md:44]: Did you actually get this stat for us?

File: SEP-0011.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+- Developer documentation can provide clear instructions to contributors on how to implement *Black* formatting in their development process.
+  Issues with this process are no more difficult (and arguably easier) to resolve than enforcing PEP-8 standards.
+
+Therefore, this SEP proposes to adopt *Black* for the SunPy Project.
+
+#### Concerns and mitigations
+
+##### Foolish consistency
+
+A concern that has been highlighted by some members of the community (including, initially, the author of this SEP, and highlighted in detail in [this blog post](https://luminousmen.com/post/my-unpopular-opinion-about-black-code-formatter) is that adopting *Black* violates the PEP-8 principle that "A Foolish Consistency is the Hobgoblin of Little Minds".
```
Comment (Cadair) [inline SEP-0011.md:76]: you really need to edit this to not just be APE 20 :stuck_out_tongue:

File: SEP-0011.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    matrix = [[1, 0, 0],
+              [0, 1, 0],
+              [0, 0, 1]]
+```
+
+*Black* reformats this as shown below in a way that is very different from the typical representation for a matrix and makes it harder to read the structure
+
+```python
+    matrix = [[1, 0, 0], [0, 1, 0], [0, 0, 1]]
+```
```
Comment (Cadair) [inline SEP-0011.md:97]: Entertainingly this example will be formatted differently if you include a trailing comma on the last row.

File: SEP-0011.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+Enthusiastic champions of the effort will largely do the work, but we recognize and appreciate that additional effort will be required from maintainers that are neutral or against the adoption of *Black*.
+
+#### Backports / release branches
+
+All will be *Black* to ensure that backports are easy to do.
+
+#### New contributors
+
+New contributors may be more discouraged by more steps.
+However, this *replaces* the existing PEP-8 check, so that it is not "another step".
```
Comment (Cadair) [inline SEP-0011.md:114]: I think this section really needs rewriting to be more sunpy appropriate, given we already have pre-commit etc.

File: SEP-0011.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+| author(s)     | Nabil Freij                                             |
+| contact email | nabil.freij@gmail.com                                   |
+| date-creation | 2022-10-25                                              |
+| type          | standard                                                |
+| discussion    | Loved it                                                |
+| status        | accepted                                                |
+
+## Introduction
+
+This SEP proposes that all packages under the SunPy organization adopt [Black](https://github.com/psf/black) for formatting code.
```
Comment (nabobalis) [inline SEP-0011.md:15]:
```suggestion
This SEP proposes that all packages under the SunPy organization adopt [Black](https://github.com/psf/black) for formatting code.
This is basically a copy and paste of the [Astropy APE 20](https://github.com/astropy/astropy-APEs/blob/main/APE20.rst) with astropy -> SunPy project.
```

File: SEP-0011.md (hunk 1 above)
Comment (nabobalis) [reply SEP-0011.md:1]: Done

**Review by Cadair — COMMENTED** [review]
I think we really need something more nuanced to start a meaningful discussion than the contents of APE 0 run through sed. I have left a few comments.

File: SEP-0011.md (hunk 2 above)
Comment (nabobalis) [reply SEP-0011.md:44]: No but its probably higher for sunpy.

File: SEP-0011.md (hunk 3 above)
Comment (nabobalis) [reply SEP-0011.md:76]: I can delete this?

**Comment by Cadair** [comment]
I think largely that's true, but the massive obvious thing this draft completely ignores the experience of us running black on a bunch of packages already, and makes no mention of the discussions we have had over the years. Also it's *really* long.

**Comment by nabobalis** [comment]
> I think largely that's true, but the massive obvious thing this draft completely ignores the experience of us running black on a bunch of packages already, and makes no mention of the discussions we have had over the years. Also it's _really_ long.

I can make it shorter.

File: SEP-0011.md (hunk 5 above)
Comment (nabobalis) [reply SEP-0011.md]: Done

File: SEP-0011.md (hunk 4 above)
Comment (nabobalis) [reply SEP-0011.md]: Removed.

File: SEP-0011.md (hunk 3 above)
Comment (nabobalis) [reply SEP-0011.md]: Removed.

File: SEP-0011.md (hunk 2 above)
Comment (nabobalis) [reply SEP-0011.md]: Updated.

**Review by dstansby — APPROVED** [review]
👍 I am pro doing this, and think the SEP sounds sensible

**Comment by nabobalis** [comment]
Replaced by #77

**Pull request closed without merging**